test/dtls_mtu_test.c

```diff
@@ -229,6 +229,18 @@ static int test_server_mtu_larger_than_max_fragment_length(void)
                                       NULL, NULL)))
         goto end;
 
+#ifndef OPENSSL_NO_SCTP
```

Comment:
When sctp is enabled `ossl_statem_export_allowed()` gets called through `SSL_export_keying_material()` -> `tls13_export_keying_material()` when sending/receiving ServerHello and the ClientKeyExchange.

As Matt notes in https://github.com/openssl/openssl/pull/24426#issuecomment-2137382145 SCTP support in DTLSv1.3 is TBD. So instead of disabling the failing test completely I set the maximum version to DTLS1.2 to test for regressions.

Reply:
But maybe we should wait for https://github.com/openssl/openssl/pull/24605 to be merged and then I can rebase on top of that to revert the change completely.